src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/PropertyAccessor.cs

```diff
@@ -3,21 +3,34 @@
 using System.Reflection;
 using System.Reflection.Emit;
 using System.Runtime.CompilerServices;
+using Jint.Native;
+using Jint.Native.Object;
 using Microsoft.CSharp.RuntimeBinder;
+using Raven.Server.Documents.Patch;
 
 namespace Raven.Server.Documents.Indexes.Persistence.Lucene.Documents
 {
     public delegate object DynamicGetter(object target);
 
-    public class PropertyAccessor
+    public class PropertyAccessor : IPropertyAccessor
     {
-        public readonly Dictionary<string, Accessor> Properties = new Dictionary<string, Accessor>();
+        private readonly Dictionary<string, Accessor> Properties = new Dictionary<string, Accessor>();
 
-        public readonly List<KeyValuePair<string, Accessor>> PropertiesInOrder =
+        private readonly List<KeyValuePair<string, Accessor>> propertiesInOrder =
             new List<KeyValuePair<string, Accessor>>();
 
-        public static PropertyAccessor Create(Type type)
+        public IEnumerable<(string Key, object Value, bool IsGroupByField)> GetPropertiesInOrder(object target)
         {
+            foreach ((var key, var value) in propertiesInOrder)
+            {
+                yield return (key, value.GetValue(target), value.IsGroupByField);
```

Comment:
Do we really need to return the actual value here (`value.GetValue(target)`)? This forces us to do `foreach` here on every call. We could introduce the abstraction over the `Accessor` and just return `propertiesInOrder` collection.